services/base-service/models/expert/api.js

```diff
@@ -279,47 +168,28 @@ router.route(
   }
 );
 
+function expert_valid_path(options) {
+  const def = {
+    "description": "Get an expert by id",
+    "parameters": [
+      "#components/parameters/expertId"
+    ],
+    "responses": {
+      "Expert": "#/components/responses/Expert",
+      "$ref": "#/components/responses/404_Expert_not_found"
+    }
+  };
+  const this_path=openapi.validPath({...def, ...options})
+  return this_path;
+//  return (req, res,next) => {
```

Comment:
@qjhart this seems to be because `validPath()` is a function that returns another function. So the validation actually occurs when you add it as a middleware function because it executes at that point, with the req/res/next being in scope.

That function looks like this, for reference:
```
middleware.validPath = function (schema = {}) {
    let validate
    function validSchemaMiddleware (req, res, next) {
      if (!validate) {
        validate = makeValidator(middleware, getSchema(validSchemaMiddleware), opts)
      }
      return validate(req, res, next)
    }

    setSchema(validSchemaMiddleware, schema)
    return validSchemaMiddleware
  }
```

We can call it by passing req/res/next into the `expert_valid_path(req, res, next, options)` function, and then calling the validation function at the end of that function like so-
`return openapi.validPath({...def, ...options})(req, res, next);`

Reply:
Not sure what you mean there.  The code as written replicates the validPath methodology, I want to chain that with the error checking, if that's how we want to catch validation errors

Reply:
Ah I think I see now. It looks like using parameter values like `#components/parameters/expertId` don't resolve to the actual params, so the validation runs but doesn't fail. Changing to this works though-

```
const def = {
    "description": "Get an expert by id",
    "parameters": [
      openapi?.document?.components?.parameters?.expertId,
      // openapi?.document?.components?.parameters?.fakeId
    ],
    "responses": [
      openapi?.document?.components?.responses?.Expert,
      openapi?.document?.components?.responses?.['Expert_not_found']
    ]
```